crates/evm/src/decode.rs

```diff
@@ -185,19 +185,33 @@ pub fn decode_revert(
                 }
             }
             // optimistically try to decode as string, unknown selector or `CheatcodeError`
-            String::decode(err)
-                .ok()
+            let error = DynSolType::abi_decode(&DynSolType::String, err)
+                .map_err(|_| eyre::eyre!("Could not optimisitically decode String"))
+                .and_then(|v| {
+                    v.as_str().map(|s| s.to_owned()).ok_or(eyre::eyre!("Could not decode string"))
+                })
+                .ok();
+
+            let error = error.filter(|err| err.as_str() != "");
+            error
                 .or_else(|| {
                     // try decoding as cheatcode error
                     if err.starts_with(ERROR_PREFIX.as_slice()) {
-                        String::decode(&err[ERROR_PREFIX.len()..]).ok()
+                        DynSolType::abi_decode(&DynSolType::String, &err[ERROR_PREFIX.len()..])
+                            .map_err(|_| eyre::eyre!("Could not decode cheatcode string error"))
+                            .and_then(|v| {
+                                v.as_str()
+                                    .map(|s| s.to_owned())
+                                    .wrap_err("Bad cheatcode string decode")
+                            })
+                            .ok()
```

Comment:
this API is also more verbose than before, why doesn't `String::decode` work?

Reply:
`String::decode` came from the `AbiDecode` trait—we don't have an analogous trait in Alloy. We could probably get one? cc @DaniPopes